### PR TITLE
Merge feature/init_java_app into main

### PR DESCRIPTION
**Pull request by bmarcu13**
- Created the skeleton of the Java app
- Implemented login full functionality.
- Created the skeleton for the main screen.

### CODE REVIEW

File: app/src/model/PermissionService.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package model;
+
+public class PermissionService {	
+	public static final int PERMISSION_HR = 0;
+	public static final int PERMISSION_FINANCIAL = 1;
+	public static final int PERMISSION_MEDICAL = 2;
+	
+	private static PermissionService permissionSerivce = null;
```
Comment (RaresMrc) [inline app/src/model/PermissionService.java:8]: Typo in "permissionSerivce"

File: app/src/model/PermissionService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		{
+			permissionSerivce = new PermissionService();
+		}
+		return permissionSerivce;
+	}
+	
+	public void setUserPermission(int _userPermission) throws Exception
+	{
+		if(!isInitialized)
+		{
```
Comment (RaresMrc) [inline app/src/model/PermissionService.java:28]: Set **isInitialized = true** after initializing the user's permissions.

File: app/src/view/LoginView.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	
+	private JLabel errorMessageLabel = new JLabel();
+	private JLabel titleLabel = new JLabel("Login");
+	private JLabel usernameLabel = new JLabel("Username");
+	private JLabel passwordLabel = new JLabel("Password");
+	
+	private JButton submitButton = new JButton("Login");
+	
+	public LoginView() 
+	{
```
Comment (RaresMrc) [inline app/src/view/LoginView.java:28]: Typo in "initComponen"

File: app/src/view/LoginView.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	private JPasswordField passwordTextField = new JPasswordField(20);
+	
+	private JLabel errorMessageLabel = new JLabel();
+	private JLabel titleLabel = new JLabel("Login");
+	private JLabel usernameLabel = new JLabel("Username");
+	private JLabel passwordLabel = new JLabel("Password");
+	
+	private JButton submitButton = new JButton("Login");
+	
+	public LoginView()
```
Comment (RaresMrc) [inline app/src/view/LoginView.java:27]: Typo in "initComponen"

File: app/src/view/LoginView.java
Comment (RaresMrc) [inline app/src/view/LoginView.java:1]: Typo in "initComponen" method from the LoginView class

File: app/src/view/MainView.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		pageContainer.add(FINANCIAL_TAB, financialView);
+		pageContainer.add(MEDICAL_TAB, medicalView);
+		
+		add(menuContainer, BorderLayout.NORTH);
+		add(pageContainer, BorderLayout.CENTER);
+	}
+	
+	public void switchTab(String tabName)
+	{
+		cardLayout.show(pageContainer, tabName);
```
Comment (RaresMrc) [inline app/src/view/MainView.java:59]: Could check for whether the tab you want to switch to exists.

**Review by RaresMrc — COMMENTED** [review]
Setting isInitialized = true comes first, the tab switching check is not a priority.

**Review by RaresMrc — APPROVED** [review]

File: app/src/view/MainView.java (hunk 5 above)
Comment (bmarcu13) [reply app/src/view/MainView.java:59]: This is unnecessary, as discussed.

**Pull request merged**